backend.py

```diff
@@ -60,6 +54,39 @@ def __repr__(self):
         return "<State> {} {}>".format(self.board, self.robots)
 
 
+class Direction(Enum):
+    N = (0, (0, +1))
+    E = (90, (+1, 0))
+    S = (180, (0, -1))
+    W = (270, (-1, 0))
+
+    def __new__(cls, keycode, vector):
+        """
+        Get attributes value and vector of the given Direction class values.
+
+        Override standard enum __new__ method.
+        Makes it possible to change vector when the object is rotated.
+        With degrees change (value) there comes the vector change (vector).
+        """
+        obj = object.__new__(cls)
+        obj._value_ = keycode
+        obj.vector = vector
+        return obj
```

Comment:
Gratuluju k nalezení řešení!
Je to ale opravdu více “magické” než by mohlo být. Mělo by stačit místo toho přidat `__init__`:

```python
    def __init__(self, degrees, vector):
        self.degrees = degrees
        self.vector = vector
```

Atribut `value` by pak byl nějaká interní <var>n</var>-tice, ale to nevadí – snad se shodneme, že `degrees` je stejně lepší jméno.

Reply:
Pak bych te poprosila zitra o ukazku, jak s tim dal pracovat. 
V implementaci s prejmenovanim `__init__()` jsem mela obrovsky - nevyreseny - problem s napsanim metody rotate() tak, aby vracela oba atributy.

Nyni objekt, na kterem uplatnim metodu: `Robot.Direction.N.rotate("left")` vrati `Robot.Direction.W` s korektnimi koordinaty instance Direction.W a tile_property (nove pridanym).

V pripade implementace vyse dopadlo mi to takto (verim, ze muze za tim byt jen nejake nedopatreni, takze v pripade zprovozneni klidne to prepisu):
```
Traceback (most recent call last):
  File "numer.py", line 38, in <module>
    aa = Direction.E.rotate("left")
  File "numer.py", line 22, in rotate
    return Direction((self.degrees + 270) % 360)
  File "/usr/lib/python3.6/enum.py", line 293, in __call__
    return cls.__new__(cls, value)
  File "/usr/lib/python3.6/enum.py", line 535, in __new__
    return cls._missing_(value)
  File "/usr/lib/python3.6/enum.py", line 548, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 0 is not a valid Direction
```
Problem je, ze pokud z metody rotate() vratim jen jeden argument, chybi druhy. Nedokazu ale dostat se k vypocitani korektniho druheho atributu (neco jako: pokud vysledkem rotace je objekt se smerem 0 degrees, pak mi najdi druhy atribut tohoto objektu a vrat mi novy validni objekt tridy Direction).
+ Odkaz pridam